pkgs/applications/misc/plover/default.nix

```diff
@@ -52,4 +69,45 @@
       makeWrapperArgs+=("''${qtWrapperArgs[@]}")
     '';
   };
+
+  plugins-manager = with python36Packages; buildPythonPackage rec {
+    pname = "plover-plugins-manager";
+    version = "0.5.16";
+
+    meta = with lib; {
+      description = "OpenSteno Plover stenography software plugin manager";
+      maintainers = with maintainers; [ tckmn ];
+      license     = licenses.gpl2;
+    };
+
+    src = fetchFromGitHub {
+      owner = "benoit-pierre";
+      repo = "plover_plugins_manager";
+      rev = "7f697da4a97cf92eeb639742a115a89887155c90";
+      sha256 = "152c02anzzvk2a5mgsnlmc1bc48lhgqf58rw2z6ir1x5carpzm91";
+    };
+
+    # apply nix sys.path wrapper to plugin manager's python call
+    postPatch = "patch -p1 <${./plugins-manager.patch}";
+
+    # tests try to instantiate a virtualenv and lack permission
+    doCheck = false;
+
+    propagatedBuildInputs = [ pip pkginfo dev pygments readme_renderer requests requests-cache requests-futures setuptools wheel ];
+  };
+
+  dev-with-plugins = dev.overrideAttrs (old: {
```

Comment:
Couple of thoughts:

Recommend adding a `withPluginManager` passthrough to the `dev` attribute and having this point to it. This allows one to override `plover.dev` if they desire and still have access to `withPluginManager` behavior without having to create an overlay (through `(plover.dev.overrideAttrs (old: ...)).withPluginManager`)

Also I think it's probably confusing to have a `withPackages`-sounding attribute in the packageset be an attribute that doesn't take an argument (namely, a list of the plugins). I notice that you used kebab-case to name it, which disambiguates it a little from functions like `emacsWithPackages` but personally I'd like to see 1 of 3 things:
  - Making it crystal clear exactly what this is -- ie. calling it something like `dev-with-plugin-manager`
  - Just optionally adding `plugin-manager` to dev (make it configurable with an override)
  - Implementing `emacsWithPackages`-like functionality

Reply:
I've distilled out my thoughts on this and I'm pretty sure the second option is the right way to do this. Plover coming with the plugin manager is *expected behavior* when installing Plover, so it's reasonable to have it be the default.

Ideally it would be disable-able by overriding the package for those who don't want it, and the third option (`emacsWithPackages`-like functionality) could be implemented later by anyone who felt inclined to do so.